### ISSUE TITLE
Destroy waits indefinitely on a resource that can't be destroyed

### ISSUE THREAD

**Issue by sabrehagen** [hashicorp/terraform#11127]
### Terraform Version
v0.8.2

### Affected Resource(s)
- aws_vpc

### Debug Output
https://gist.github.com/sabrehagen/63971123dd5c4b65c7cbb3e8c45ff475

### Expected Behavior
Program should exit and alert the user the resource can't be destroyed.

### Actual Behavior
Terraform waits indefinitely on a resource that can't be destroyed.

```
aws_vpc.default: Destroying...
aws_vpc.default: Still destroying... (10s elapsed)
aws_vpc.default: Still destroying... (20s elapsed)
aws_vpc.default: Still destroying... (30s elapsed)
aws_vpc.default: Still destroying... (40s elapsed)
aws_vpc.default: Still destroying... (50s elapsed)
aws_vpc.default: Still destroying... (1m0s elapsed)
aws_vpc.default: Still destroying... (1m10s elapsed)
```

### Important Factoids
This occurred after a failed plan where subnet cidrs were in conflict and couldn't be created. Command sequence was:

    terraform apply (failed => fixed terraform config)
    terraform apply (vpc cidr changed so attempt to destroy)

**Comment by grubernaut** [comment]
Hi @sabrehagen, thanks for the issue. 

Do you recall if the destroy error'd out with a specific error, or did you stop the process manually? 

The debug output shows the dependency violation on the attempted VPC destroy. The dependency error could be resolved in the span of five minutes, it depends on the configuration used.

We retry the destruction of VPCs for up to 5 minutes, to allow time for the destruction of any dependent resources complete and to get cleaned up fully. Have you experienced times longer than the 5 minutes? Terraform should abort the attempt and log the error after 5 minutes. 

Do you have any sample configuration that can be used to fully reproduce your error as well?
Thanks!

**Comment by sabrehagen** [comment]
Hi,

The create errored out, but I don't have the error it failed with. The destroy I killed though after ~3 minutes of waiting. I didn't wait the full 5 minutes as I assumed it was in an infinite loop.

I don't have the same config that produced the error unfortunately as the cidr block issue has been resolved days ago. I can't be much more help from here. Sounds like you handle the case at the 5 minute mark though.
